packages/fxa-payments-server/src/components/Coupon/index.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {
+  useState,
+  useContext,
+  MouseEventHandler,
+  FormEventHandler,
+} from 'react';
+import { Localized } from '@fluent/react';
+import './index.scss';
+import { CouponContext } from '../../lib/CouponContext';
+
+export const Coupon = () => {
+  const [hasCoupon, setHasCoupon] = useState(false);
+  const [couponCode, setCouponCode] = useState('');
```

Comment:
Having two states and manually set them feels fragile.  The hasCoupon state is dependent on the value of the couponCode, but the consts are unrelated.

Reply:
One is holding the state of the coupon code as it is entered into the input field, the other is set/reset depending on if the coupon code was valid after hitting apply and manages the UI. Right now it is mainly used for handling the dummy data and I can probably rip it out afterwards, but I am not sure how the 2 might be consolidated.

Reply:
> One is holding the state of the coupon code as it is entered into the input field, the other is set/reset depending on if the coupon code was valid after hitting apply and manages the UI.

Ah, sorry, I misunderstood then.  Thanks for the clarification.